core/src/main/scala/kafka/server/FetchSession.scala

```diff
@@ -187,24 +205,86 @@ class CachedPartition(val topic: String,
   }
 }
 
+/**
+ * Very similar to CachedPartition above, CachedUnresolvedPartition is used for incremental fetch requests.
+ * These objects store partitions that had topic IDs that could not be resolved by the broker.
+ *
+ * Upon each incremental request in the session, these partitions will be loaded. They can either be removed
+ * through resolving the partition with the broker's topicNames map or by receiving an unresolved toForget ID.
+ *
+ * Since these partitions signify an error, they will always be returned in the response.
+ */
+
+class CachedUnresolvedPartition(val topicId: Uuid,
+                      val partition: Int,
+                      var maxBytes: Int,
+                      var fetchOffset: Long,
+                      var leaderEpoch: Optional[Integer],
+                      var fetcherLogStartOffset: Long,
+                      var lastFetchedEpoch: Optional[Integer]) {
```

Comment:
Does an unresolved partition have all these fields populated? Or do we have it here because the topic may be resolved later?

Reply:
We need to keep the data from the fetch request for when we resolve the partition.